packages/database/src/TupaiaDatabase.js

```diff
@@ -683,26 +682,13 @@ function addJoin(baseQuery, recordType, joinOptions) {
 }
 
 function getColSelector(connection, inputColStr) {
-  const regexp = new RegExp(/->>?/, 'g');
-  if (regexp.test(inputColStr)) {
-    // Shorthand way of querying json property
-    // TODO: Replace with knex json where functions, eg. whereJsonPath
-    const [first, ...rest] = inputColStr.split(regexp);
-    if (rest.length === 1) {
-      // get the first separator (there will only be 1 in this case)
-      const separator = inputColStr.match(regexp)[0];
-      // e.g. 'config->>colour' is converted to config->>'colour' and 'config->colour' is converted to config->'colour'
-      return connection.raw(`??${separator}?`, [first, ...rest]);
-    }
-    // e.g. 'config->item->>colour' is converted to config->'item'->>'colour'
-    const last = rest.slice(-1);
-    const middle = rest.slice(0, rest.length - 1);
-    return connection.raw(`??->${middle.map(i => '?').join('->')}->>?`, [
-      first,
-      ...middle,
-      ...last,
-    ]);
-  }
+  const jsonOperatorPattern = /->>?/g;
+  if (!jsonOperatorPattern.test(inputColStr)) return inputColStr;
+
+  const params = inputColStr.split(jsonOperatorPattern);
+  const allButFirst = params.slice(1);
 
-  return inputColStr;
+  // Turn `config->item->>colour` into `config #>> '{item,colour}'`
+  // For some reason, Knex fails when we try to convert it to `config->'item'->>'colour'`
+  return connection.raw(`?? #>> '{${allButFirst.map(() => '?').join(',')}}'`, params);
```

Comment:
Looks like the tests fail due to: 
```
error: select "entity"."id", "entity"."code", "entity"."parent_id", "entity"."name", "entity"."type", ST_AsGeoJSON("entity"."point") as "point", ST_AsGeoJSON("entity"."region") as "region", "entity"."image_url", "entity"."country_code", ST_AsGeoJSON("entity"."bounds") as "bounds", "entity"."metadata", "entity"."attributes" from "entity" where "id" = $1 and "attributes" #>> '{$2}' = $3 and "attributes" #>> '{$4}' in ($5, $6) and "attributes" #>> '{$7}' = $8 limit $9 - could not determine data type of parameter $2
```

Possibly a typecast could fix this, eg.
```suggestion
  return connection.raw(`?? #>> '{${allButFirst.map(() => '?').join(',')}}'::text[]`, params);
```

Reply:
Hm, thanks for looking into this, though that doesn’t seem to fix it. Also tried:

```ts
return connection.raw(`?? #>> '{${allButFirst.map(() => '?::text').join(',')}}'`, params);
```

but it throws the same error.

Just to be certain, I also reverted to using the rest operator and the issue persists. It’s definitely something to do with parameterising the query. I’ll keep investigating